server/http/src/service.rs

```diff
@@ -225,27 +266,35 @@ where
                         .run_query(query)
                         .map_err(|e| GraphQLServerError::from(e))
                 })
-                .then(move |result| {
-                    let elapsed = start.elapsed().as_millis();
-                    match result {
-                        Ok(_) => info!(
-                            logger,
-                            "GraphQL query served";
-                            "subgraph_deployment" => sd_id.deref(),
-                            "query_time_ms" => elapsed,
-                            "code" => LogCode::GraphQlQuerySuccess,
-                        ),
-                        Err(ref e) => error!(
-                            logger,
-                            "GraphQL query failed";
-                            "subgraph_deployment" => sd_id.deref(),
-                            "error" => e.to_string(),
-                            "query_time_ms" => elapsed,
-                            "code" => LogCode::GraphQlQueryFailure,
-                        ),
-                    }
-                    GraphQLResponse::new(result)
-                }),
+                .then(
+                    move |result: Result<QueryResult, GraphQLServerError>| -> GraphQLResponse {
+                        let elapsed = start.elapsed();
+                        match result {
+                            Ok(_) => {
+                                service_metrics.observe_query_execution_time(
+                                    elapsed.as_secs_f64(),
+                                    sd_id.deref().to_string(),
+                                );
+                                info!(
+                                    logger,
+                                    "GraphQL query served";
+                                    "subgraph_deployment" => sd_id.deref(),
+                                    "query_time_ms" => elapsed.as_millis(),
+                                    "code" => LogCode::GraphQlQuerySuccess,
+                                )
+                            }
+                            Err(ref e) => error!(
+                                logger,
+                                "GraphQL query failed";
+                                "subgraph_deployment" => sd_id.deref(),
```

Comment:
Is this not also worth measuring the time for?

Reply:
Definitely. I had questions about how we want to handle this, so I started with the simplest case.  The error case could return early from a bad request and I don't want those times to skew the aggregate numbers.  

How do you feel about keeping `subgraph_query_execution_time` (or `subgraph_query_time`) for successful queries and add another histogram called `subgraph_failed_query_execution_time`?

Reply:
`subgraph_query_time` + `subgraph_failed_query_time`? Sounds good to me 😄